src/core/reducers/data/navigations.ts

```diff
@@ -105,20 +95,23 @@ export const deselectRefinement = (state: State, { navigationId, index: refineme
   }
 };
 
-// tslint:disable-next-line max-line-length
-export const addRefinementByIndex = (state: State, { navigationId, index: refinementIndex }: Actions.Payload.Search) => {
-
-  return {
-    ...state,
-    byId: {
-      ...state.byId,
-      [navigationId]: {
-        ...state.byId[navigationId],
-        selected: [refinementIndex],
-      },
-    },
-  };
-};
+const generateNavigationId = (state: State, navigationId: string, refinement: any, index: number) => ({
```

Comment:
I think this would be better named `generateNavigation()` as that is what it does, rather than generate a `navigationId`